sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/models/QueueMessageDecodingFailure.java

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.queue.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.storage.queue.QueueAsyncClient;
+
+import java.util.Objects;
+
+/**
+ * Contains information about message that could not be decoded.
+ */
+@Immutable
+public class QueueMessageDecodingFailure {
```

Comment:
Make this `final`. Also, consider naming this type as `QueueMessageDecodingError`.